GUI/CustomFragmentationWindow.xaml.cs

```diff
@@ -0,0 +1,61 @@
+﻿using EngineLayer;
+using Nett;
+using Proteomics.Fragmentation;
+using System.Collections.Generic;
+using System.Windows;
+using TaskLayer;
+using System.IO;
+using System.ComponentModel;
+
+namespace MetaMorpheusGUI
+{
+    /// <summary>
+    /// Interaction logic for CustomFragmentationIons.xaml
+    /// </summary>
+    public partial class CustomFragmentationWindow : Window
+    {
+        private CustomFragmentation CustomFragmentation = new CustomFragmentation();
+
+        public CustomFragmentationWindow() : this(null)
+        {
+        }
+
+        public CustomFragmentationWindow(List<ProductType> list)
+        {
+            InitializeComponent();
+            // update checkboxes
+            if (list != null)
+            {
+                bCheckBox.IsChecked = list.Contains(ProductType.b);
```

Comment:
this PR works but I think it can be more concise and a bit more flexible:
1. Instead of reading checkboxes to a list, then storing the list to a toml, then reading the toml to another list, can you just directly go from checkboxes -> add the `ProductType`s to the `DissociationType.ProductsFromDissociationType[Custom]` list? then in the common params constructor, you can just set `CustomIons = DissociationType.ProductsFromDissociationType[Custom]` without needing to worry about modifying the constructor parameters, reading/writing .tomls, etc.
2. Instead of hard-coding checkboxes for b/y/c/z ions, I'd like a dynamic list of checkboxes that expands to include all known product types. `DissociationTypeCollection.ProductsFromDissociationType.Values.SelectMany(p => p).Distinct()` will get you all known product types.
3. Need a unit test to ensure custom product types are handled correctly between tasks. So if task1 uses b/y ions, and task2 uses c/z ions, and task3 uses y/c/z ions, make sure these are read and searched correctly from the .tomls

Reply:
ok, working on it